packages/react-composites/src/composites/CallComposite/adapter/AzureCommunicationCallAdapter.ts

```diff
@@ -211,58 +246,105 @@ export class AzureCommunicationCallAdapter implements CallAdapter {
   }
 
   public async createStreamView(remoteUserId?: string, options?: VideoStreamOptions): Promise<void> {
-    if (remoteUserId === undefined) {
-      await this.handlers.onCreateLocalStreamView(options);
-    } else {
-      await this.handlers.onCreateRemoteStreamView(remoteUserId, options);
+    try {
+      if (remoteUserId === undefined) {
+        await this.handlers.onCreateLocalStreamView(options);
+      } else {
+        await this.handlers.onCreateRemoteStreamView(remoteUserId, options);
+      }
+    } catch (e) {
+      this.context.setError(e);
+      this.emitter.emit('error', e);
     }
   }
 
   public async disposeStreamView(remoteUserId?: string): Promise<void> {
-    if (remoteUserId === undefined) {
-      await this.handlers.onDisposeLocalStreamView();
-    } else {
-      await this.handlers.onDisposeRemoteStreamView(remoteUserId);
+    try {
+      if (remoteUserId === undefined) {
+        await this.handlers.onDisposeLocalStreamView();
+      } else {
+        await this.handlers.onDisposeRemoteStreamView(remoteUserId);
+      }
+    } catch (e) {
+      this.context.setError(e);
+      this.emitter.emit('error', e);
     }
   }
 
   public async leaveCall(): Promise<void> {
     const callId = this.call?.id;
-    await this.handlers.onHangUp();
+    this.stopCamera();
+    this.mute();
+
+    try {
+      await this.handlers.onHangUp();
+    } catch (e) {
+      this.context.setError(e);
```

Comment:
Out of interest, how is the context error used currently? Do we make use of it or is it in the state for a user to query? I'm worried that because we call setError on every error that an error a user wants to act on will be overwritten before they get the chance to?

Reply:
Right, that is a valid concern. One way to not lose the error is to have Contoso register the error listener and they will be able to receive all errors without missing any. @PorterNan , is the original design for state to hold the error so it can be retrieved? If so what do you think of James's concern about the error being overwritten with a new error?

Reply:
Approving so this comment isn't blocking - but still wondering if we'll run into a problem with errors stomping over each other

Reply:
Synced up with Nan and James offline. Looks like initially we are planning to have one error in the context and this is shown in the error bar. For the one error case, it is fine to just show the latest error. In the future the design might change to allow for some history of errors but for now I'll consider this resolved as it seems to be the current design but please let me know if there is any concerns.

Reply:
Let's rename the state field to be `lastError` to better capture the semantics.
We should also provide a `clearLastError()` method in the stateful client to allow Contoso (or the error bar `onClose` handler) to clear the error.

Reply:
The rename doesn't have to be in this PR.

Reply:
Going to leave that for another PR as it seems to be a feature add.